src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -813,19 +809,19 @@ def can_clause_dependent_columns_flatten(
     return can_be_flattened
 
 
-def initiate_column_states(column_names: List[str]) -> ColumnStateDict:
+def initiate_column_states(column_attrs: List[Attribute]) -> ColumnStateDict:
     column_states = ColumnStateDict()
-    for name in column_names:
-        column_states[name] = ColumnState(
-            name,
+    for attr in column_attrs:
+        column_states[attr.name] = ColumnState(
+            attr.name,
```

Comment:
We should use `analyzer.analize(attr)` instead of using `attr.name`. Or else `column_states` will have the wrong column name.